queue_services/entity-filer/tests/unit/test_worker/test_correction_ben.py

```diff
@@ -0,0 +1,768 @@
+    mocker.patch('entity_filer.filing_processors.filing_components.business_profile.update_business_profile',
+                 return_value=None)
+    mocker.patch('legal_api.services.bootstrap.AccountService.update_entity', return_value=None)
+
+    # Test
+    with patch.object(NaicsService, 'find_by_code', return_value=naics_response):
+        await process_filing(filing_msg, app)
+
+    # Check outcome
+    business = Business.find_by_internal_id(business_id)
+
+    resolution_dates = [res.resolution_date for res in business.resolutions.all()]
+    if test_name == 'ben_add_resolution_dates':
+        assert len(business.resolutions.all()) == 3
+        assert resolution_dates[0] == parse(resolution_dates_json1).date()
+        assert resolution_dates[1] == parse(resolution_dates_json2).date()
+        assert resolution_dates[2] == parse(new_resolution_dates).date()
+
+    if test_name == 'ben_update_existing_resolution_dates':
+        assert len(business.share_classes.all()) == 2
+        assert parse(resolution_dates_json1).date() in resolution_dates
+        assert parse(updated_resolution_dates).date() in resolution_dates
+        assert parse(resolution_dates_json2).date() not in resolution_dates
+
+    if test_name == 'ben_update_with_new_resolution_dates':
+        assert len(business.resolutions.all()) == 1
+        assert parse(updated_resolution_dates).date() in resolution_dates
+        assert parse(resolution_dates_json1).date() not in resolution_dates
+        assert parse(resolution_dates_json2).date() not in resolution_dates
+
+    if test_name == 'ben_delete_resolution_dates':
+        assert len(business.resolutions.all()) == 1
+        assert resolution_dates[0] == parse(resolution_dates_json2).date()
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    'test_name',
+    [
+        'ben_add_share_class',
+        'ben_update_existing_share_class',
+        'ben_update_with_new_share_class',
+        'ben_delete_share_class',
+    ]
+)
+async def test_worker_share_class_and_series_change(app, session, mocker, test_name):
+    """Assert the worker processes the court order correctly."""
+    identifier = 'BC1234567'
+    business = create_entity(identifier, 'BEN', 'Test Entity')
+    business_id = business.id
+
+    share_class_json1 = BEN_CORRECTION['filing']['correction']['shareStructure']['shareClasses'][0]
+    share_class_json2 = BEN_CORRECTION['filing']['correction']['shareStructure']['shareClasses'][1]
+
+    filing = copy.deepcopy(BEN_CORRECTION)
+
+    corrected_filing = factory_completed_filing(business, BEN_CORRECTION_APPLICATION)
+    filing['filing']['correction']['correctedFilingId'] = corrected_filing.id
+
+    filing['filing']['correction']['contactPoint'] = CONTACT_POINT
+
+    if test_name == 'ben_add_share_class':
+        new_share_class_json = BEN_CORRECTION['filing']['correction']['shareStructure']['shareClasses'][1]
+        new_share_class_json['id'] = 3
```

Comment:
if it's a new share class being added, I don't think the id property will exist

Reply:
Fixed.